test/test_in_toto_record.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+
+"""
+<Program Name>
+  test_in_toto_record.py
+
+<Author>
+  Lukas Puehringer <lukas.puehringer@nyu.edu>
+
+<Started>
+  Nov 28, 2016
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  Test in_toto_record command line tool.
+
+"""
+
+import os
+import sys
+import unittest
+import logging
+import argparse
+from mock import patch
+
+from toto.util import generate_and_write_rsa_keypair
+from toto.models.link import Link
+from toto.in_toto_record import main as in_toto_record_main
+from toto.in_toto_record import in_toto_record_start, in_toto_record_stop
+
+# Suppress all the user feedback that we print using a base logger
+logging.getLogger().setLevel(logging.CRITICAL)
+
+def _try_remove_files(file_list):
+  if not isinstance(file_list, list):
+    return
+  else:
+    for path in file_list:
+      try:
+        os.remove(path)
+      except Exception, e:
+        pass
+
+
+class TestInTotoRecordMain(unittest.TestCase):
+  """Test in_toto_record's main() - requires sys.argv patching. """
+
+  @classmethod
+  def setUpClass(self):
+    """Generate key pair, dummy artifact and base arguments. """
+    generate_and_write_rsa_keypair("test-key")
+    self.args = [
+        "in_toto_record.py",
+        "--step-name",
+        "test-step",
+        "--key",
+        "test-key"
+    ]
+    self.test_artifact = "test-artifact"
+    open(self.test_artifact, "a").close()
+
+  @classmethod
+  def tearDownClass(self):
+    """Remove key pair, dummy artifact and link metadata. """
+    files_to_remove = [
+      "test-key", "test-key.pub", "test-step.link", self.test_artifact
+    ]
+    _try_remove_files(files_to_remove)
+
+
+  def test_command_start_stop_required_args(self):
+    """Test CLI command record start/stop with required arguments. """
+    with patch.object(sys, 'argv',  self.args + ["start"]):
+      in_toto_record_main()
+
+    with patch.object(sys, 'argv', self.args + ["stop"]):
+      in_toto_record_main()
+
+  def test_command_start_stop_optional_args(self):
+    """Test CLI command record start/stop with optional arguments. """
+    with patch.object(sys, 'argv', self.args + ["start", "--materials",
+        self.test_artifact]):
+      in_toto_record_main()
+
+    with patch.object(sys, 'argv', self.args + ["stop", "--products",
+        self.test_artifact]):
+      in_toto_record_main()
```

Comment:
The reason why I use `mock` here is that I want to test the argparser part of the cli tool.